samples/snippets/quickstart.py

```diff
@@ -0,0 +1,246 @@
+
+    print(f"Deleting {machine_name} from {zone}...")
+    operation = instance_client.delete(
+        project=project, zone=zone, instance=machine_name
+    )
+    operation = wait_for_operation(operation, project)
+    if operation.error:
+        pass
+    if operation.warnings:
+        pass
+    print(f"Instance {machine_name} deleted.")
+    return
+
+
+# [END compute_instances_delete]
+
+
+# [START compute_instances_operation_check]
+def wait_for_operation(operation: gce.Operation, project: str) -> gce.Operation:
+    """
+    This method waits for an operation to be completed. Calling this function
+    will block until the operation is finished.
+
+    Args:
+        operation: The Operation object representing the operation you want to
+            wait on.
+        project: Name of the project owning the operation.
+
+    Returns:
+        Finished Operation object.
+    """
+    kwargs = {"project": project, "operation": operation.name}
+    if operation.zone:
+        client = gce.ZoneOperationsClient()
+        # Operation.zone is a full URL address of a zone, so we need to extract just the name
+        kwargs["zone"] = operation.zone.rsplit("/", maxsplit=1)[1]
+    elif operation.region:
+        client = gce.RegionOperationsClient()
+        # Operation.region is a full URL address of a zone, so we need to extract just the name
+        kwargs["region"] = operation.region.rsplit("/", maxsplit=1)[1]
+    else:
+        client = gce.GlobalOperationsClient()
+    return client.wait(**kwargs)
+
+
+# [END compute_instances_operation_check]
+
+
+def main(project: str, zone: str, machine_name: str) -> None:
+    # You can find the list of available machine types using:
+    # https://cloud.google.com/sdk/gcloud/reference/compute/machine-types/list
+    machine_type = f"zones/{zone}/machineTypes/f1-micro"
+    # You can check the list of available public images using:
+    # gcloud compute images list
+    source_image = "projects/debian-cloud/global/images/family/debian-10"
+
+    create_instance(project, zone, machine_type, machine_name, source_image)
+
+    zone_instances = list_instances(project, zone)
+    print(f"Instances found in {zone}:", ", ".join(i.name for i in zone_instances))
+
+    all_instances = list_all_instances(project)
+    print(f"Instances found in project {project}:")
+    for i_zone, instances in all_instances.items():
+        print(f"{i_zone}:", ", ".join(i.name for i in instances))
+
+    delete_instance(project, zone, machine_name)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description=__doc__, formatter_class=argparse.RawDescriptionHelpFormatter,
+    )
+    parser.add_argument("project_id", help="Google Cloud project ID")
+    parser.add_argument("zone", help="Google Cloud zone name")
+    parser.add_argument("machine_name", help="Name for the demo machine")
+
+    args = parser.parse_args()
+
+    main(args.project_id, args.zone, args.machine_name)
```

Comment:
We have argparse in a lot of older samples but it is something we're trying to move away from (since this section of the sample file cannot be easily tested). If there's isn't specific reason this is valuable for the compute samples please remove it.